subiquity/server/controllers/install.py

```diff
@@ -306,6 +306,36 @@ async def run_curtin_step(name, stages, step_config, source=None):
                 source=source,
                 )
             await self.setup_target(context=context)
+
+            # For OEM, we basically mimic what ubuntu-drivers does:
+            # 1. Install each package with apt-get install
+            # 2. For each package, run apt-get update using only the source
+            # installed by said package.
+            # 3. Run apt-get install again for each package. This will upgrade
+            # them to the version found in the OEM archive.
+
+            # NOTE In ubuntu-drivers, this is done in a single call to apt-get
+            # install.
+            for pkg in self.model.oem.metapkgs:
+                await self.install_package(package=pkg)
+
+            if self.model.network.has_network:
```

Comment:
If someone is offline (as in an unfortunately timed network drop), either temporarily or (from the point of view of the install) permanently, what is the desired behavior?

Reply:
For the general ISO, I doubt we'll include metapackages in the pool, so they won't get suggested. For an OEM image, all the relevant metapackages and their dependencies will be in the pool, so installing them will install all the relevant drivers. It will also install the sources.list.d entry, so we shouldn't run apt-get update as it will fail (hmm... I wonder if that will be a problem the way things are today?)

This is all a bit more complicated than I thought when starting this reply so maybe it should go in a comment somewhere :-)

Reply:
I'll add a comment regarding the pool.

If we're temporarily offline, then there will be retries when trying to download the required packages - and then we will bail. However, I haven't implemented a retry for the apt-get call - and there isn't any in the `ubuntu-drivers install` command either.